dns.go

```diff
@@ -41,20 +43,34 @@ func (d *customDialer) archiveDNS(address string) (resolvedIP net.IP, err error)
 	var ipv4, ipv6 net.IP
 	var errA, errAAAA error
 
-	wg.Add(2)
+	if len(d.DNSConfig.Servers) == 0 {
+		return nil, fmt.Errorf("no DNS servers configured")
+	}
+
+	fallbackServers := min(maxFallbackDNSServers, len(d.DNSConfig.Servers)-1)
 
-	go func() {
-		defer wg.Done()
-		ipv4, errA = d.lookupIP(address, dns.TypeA)
-	}()
+	for DNSServer := 0; DNSServer <= fallbackServers; DNSServer++ {
+		if DNSServer > 0 {
+			// TODO: print warning if the default DNS server is not reachable
```

Comment:
I don't know how to log the warning here properly. Add a fmt or slog print? :(

Reply:
I guess slog is better but it will eventually get rewritten when we properly rewrite the logging..